kuwala/core/canvas/src/components/Modals/TransformationCatalog/TransformationCatalogModal.js

```diff
@@ -0,0 +1,329 @@
+        }
+    }
+
+    const renderCatalogItem = (catalogItem,index) => {
+        return (
+            <Button
+                onClick={()=>{
+                            setSelectedCatalogOption(null);
+                            setSelectedTransformationIndex(index)
+                        }}
+                selected={selectedTransformationIndex === index}
+                solid={false}
+                color={'kuwalaRed'}
+            >
+                {getCatalogItemIcon(catalogItem.icon)}
+                <span className={'font-semibold'}>
+                   {catalogItem.name}
+                </span>
+            </Button>
+        )
+    }
+
+    const CatalogBody = () => {
+        return (
+            <>
+                {
+                    selectedTransformationIndex !== null
+                    ?
+                        <SelectedTransformationCatalog/>
+                    :
+                        <div className={Classes.CatalogBodyAlertText}>
+                            <p>To pick a transformation block,</p>
+                            <p>select a transformation <span className={'text-kuwala-red'}>category</span> first</p>
+                        </div>
+                }
+            </>
+        )
+    }
+
+    const SelectedTransformationCatalog = () => {
+        if (!selectedTransformationIndex || catalogOptions.length === 0) {
+            return (
+                <div>
+                    No <span className={'text-kuwala-red'}>transformation</span> options found.
+                </div>
+            )
+        } else {
+            return (
+                <div className={Classes.SelectedTransformationContainer}>
+                    <div className={Classes.TransformationOptionsContainer}>
+                        {renderTransformationOptions(catalogOptions)}
+                    </div>
+                    <div className={Classes.SelectedOptionDetailsContainer}>
+                        <div className={Classes.OptionDetailsContainer}>
+                            {renderSelectedOptionDetails()}
+                        </div>
+                    </div>
+                </div>
+            )
+        }
+    }
+
+    const renderTransformationOptions = (options) => {
+        return options.map(renderOptionItem)
+    }
+
+    const renderOptionItem = (optionItem, index) => {
+        return (
+            <Button
+                solid={false}
+                color={'kuwalaPurple'}
+                onClick={()=>{
+                    setSelectedCatalogOption(index)
+                }}
+            >
+                <div className={Classes.WideButtonContainer}>
+                    {getCatalogItemIcon(optionItem.icon)}
+                    <span className={'font-semibold'}>
+                        {optionItem.name}
+                    </span>
+                </div>
+            </Button>
+        )
+    }
+
+    const renderSelectedOptionDetails = () => {
+        if(selectedCatalogOption === null){
```

Comment:
it was intended to be === null because it is meant to check wether or not any value is selected.

if no value is selected -> null
if any value is selected -> any numbers starting from 0

if `!selectedCatalogOption` is it will make index 0 falsey and not able to be selected.